orchagent/portsorch.cpp

```diff
@@ -1570,22 +1570,17 @@ bool PortsOrch::removePort(sai_object_id_t port_id)
 {
     SWSS_LOG_ENTER();
 
-    Port p;
-    if (getPort(port_id, p))
-    {
-        PortUpdate update = {p, false };
-        notify(SUBJECT_TYPE_PORT_CHANGE, static_cast<void *>(&update));
-    }
-
     sai_status_t status = sai_port_api->remove_port(port_id);
     if (status != SAI_STATUS_SUCCESS)
     {
-        SWSS_LOG_ERROR("Failed to remove port %" PRIx64 ", rv:%d", port_id, status);
+        if (status != SAI_STATUS_OBJECT_IN_USE)
+        {
+            SWSS_LOG_ERROR("Failed to remove port %" PRIx64 ", rv:%d", port_id, status);
+            throw runtime_error("Delete port failed");
```

Comment:
>throw runtime_error("Delete port failed"); [](start = 12, length = 42)

Treat it as an error, but not to crash the orchagent.

Reply:
Two things:
1. I did NOT want to change the existing behavior. 
2. We need to think if this can ever happen, if our theory says that it can never happen, we should throw a runtime_error.

At this point in time, if a port does NOT have any dependency, I do NOT see any reason why a removal/deletion would fail. If at all it happens, I would prefer to catch it right then and there instead of letting the system go in in-consistent state.